File: electron/protocol/paths.js
const { normalizeStyle } = require("./ir");

function defaultUpstreamPathSuffix(egressStyle, upstream) {
  const style = normalizeStyle(egressStyle);
  if (upstream?.source === "subscription:codex") return "/codex/responses";
  if (style === "claude") return "/messages";
  if (style === "openai-responses") return "/responses";
  return "/chat/completions";
}

function expectedIngressPathSuffix(ingressStyle, upstream = null) {
  return defaultUpstreamPathSuffix(ingressStyle, upstream);
}

module.exports = {
  defaultUpstreamPathSuffix,
  expectedIngressPathSuffix,
};

File: core/internal/protocol/paths.go
package protocol

import (
	"strings"

	"github.com/clovapi/switcher/internal/apistyle"
)

// DefaultUpstreamPathSuffix mirrors electron/protocol/paths.js defaultUpstreamPathSuffix.
func DefaultUpstreamPathSuffix(egress apistyle.Style, upstreamSource string) string {
	if strings.TrimSpace(upstreamSource) == "subscription:codex" {
		return "/codex/responses"
	}
	switch egress {
	case apistyle.Claude:
		return "/messages"
	case apistyle.OpenAIResponses:
		return "/responses"
	default:
		return "/chat/completions"
	}
}
